src/garage/replay_buffer/multi_task_replay_buffer.py

```diff
@@ -0,0 +1,145 @@
+"""A replay buffer memory for meta-RL."""
+
+from gym.spaces import Box, Discrete, Tuple
+import numpy as np
+
+from garage.replay_buffer.meta_replay_buffer import MetaReplayBuffer
+
+
+class MultiTaskReplayBuffer:
+    """This buffer is used in meta-RL algorithms involving multiple tasks.
+
+    It contains a list of MetaReplayBuffers that can be accessed by index.
+
+    Args:
+        max_replay_buffer_size (int): Maximum buffer size.
+        env (garage.envs.GarageEnv): Meta environment.
+        tasks (list): A list of task indices.
+
+    """
+
+    def __init__(self, max_replay_buffer_size, env, tasks):
+
+        self.env = env
+        self._obs_space = env.observation_space
+        self._action_space = env.action_space
+        self.task_buffers = dict([
+            (idx,
+             MetaReplayBuffer(
+                 max_replay_buffer_size=max_replay_buffer_size,
+                 observation_dim=get_dim(self._obs_space),
+                 action_dim=get_dim(self._action_space),
+             )) for idx in tasks
+        ])
+
+    def add_sample(self, task, observation, action, reward, terminal,
+                   next_observation):
+        """Add a sample to the buffer.
+
+        Args:
+            task (int): Task index.
+            observation (numpy.ndarrayy): Observation.
+            action (numpy.ndarray): Action.
+            reward (float): Reward.
+            terminal (bool): Terminal state.
+            next_observation (numpy.ndarray): Next obseravation.
+
+        """
+        if isinstance(self._action_space, Discrete):
+            action = np.eye(self._action_space.n)[action]
+        self.task_buffers[task].add_sample(observation, action, reward,
+                                           terminal, next_observation)
+
+    def add_path(self, task, path):
+        """Add path to a task buffer by its index.
+
+        Args:
+            task (int): Task index.
+            path (dict): Dictionary containing path information.
+
+        """
+        self.task_buffers[task].add_path(path)
+
+    def add_paths(self, task, paths):
+        """Add paths to a task buffer by its index.
+
+        Args:
+            task (int): Task index.
+            paths (dict): Dictionary containing multiple paths.
+
+        """
+        for path in paths:
+            self.task_buffers[task].add_path(path)
+
+    def terminate_episode(self, task):
+        """Terminate current episode of task buffer by its index.
+
+        Args:
+            task (int): Task index.
+
+        """
+        self.task_buffers[task].terminate_episode()
+
+    def clear_buffer(self, task):
+        """Clear a task buffer by its index.
+
+        Args:
+            task (int): Task index.
+
+        """
+        self.task_buffers[task].clear()
+
+    def random_batch(self, task, batch_size, sequence=False):
```

Comment:
What's the rationale behind this method? If seems confusing that it wraps both `random_sequence` and `random_batch` but is called `random_sequence`.

Reply:
If a recurrent encoder is used in PEARL, then `sequence` is True, and this samples trajectories instead of random transitions.

Reply:
i think that this should be a different method. methods should not change their return types based on arguments.